src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CaseReferenceValidator.java

```diff
@@ -4,40 +4,97 @@
+        EXTERNAL_CASE_REFERENCE
+    );
 
     @Nonnull
-    Validation<String, String> validate(CaseDetails theCase) {
-        return getCaseRef(theCase)
-            .flatMap(this::validateCaseRef)
+    Validation<String, String> validateAttachToCaseReference(CaseDetails theCase) {
+        return getCaseRef(theCase, ATTACH_TO_CASE_REFERENCE)
+            .map(this::validateCcdCaseRef)
             .orElseGet(() -> invalid("No case reference supplied"));
     }
 
-    private Optional<Validation<String, String>> validateCaseRef(Object caseRef) {
-        Optional<Validation<String, String>> valid = Optional.of(caseRef)
-            .filter(ref -> ref instanceof String)
-            .map(ref -> ((String) ref).replaceAll("[^0-9]", ""))
-            .map(Strings::emptyToNull)
-            .map(Validation::valid);
-        //This is done because java 8 does not have "Optional<T> orElse( Optional<T>)"
-        if (valid.isPresent()) {
-            return valid;
+    @Nonnull
+    Validation<String, String> validateSearchCaseReference(CaseDetails theCase) {
+        Validation<String, String> caseReferenceTypeValidation = validateCaseReferenceType(theCase);
+
+        if (caseReferenceTypeValidation.isValid()) {
+            Function<Object, Validation<String, String>> validationFunction =
+                CCD_CASE_REFERENCE.equals(caseReferenceTypeValidation.get())
+                    ? this::validateCcdCaseRef
+                    : this::validateExternalCaseRef;
+
+            return getCaseRef(theCase, SEARCH_CASE_REFERENCE)
+                .map(validationFunction)
+                .orElseGet(() -> invalid("No case reference supplied"));
         } else {
-            return Optional.of(invalid(format("Invalid case reference: '%s'", caseRef)));
+            return invalid("Cannot validate case reference due to the lack of valid case reference type");
         }
     }
 
-    private Optional<Object> getCaseRef(CaseDetails theCase) {
+    @Nonnull
+    Validation<String, String> validateCaseReferenceType(CaseDetails theCase) {
+        return getSearchCaseReferenceType(theCase)
+            .map(searchCaseReferenceType -> {
+                    if (VALID_CASE_REFERENCE_TYPES.contains(searchCaseReferenceType)) {
+                        return Validation.<String, String>valid((String) searchCaseReferenceType);
+                    } else {
+                        return Validation.<String, String>invalid(
+                            "Invalid case reference type supplied: " + searchCaseReferenceType
+                        );
+                    }
+                }
+            ).orElseGet(() -> invalid("No case reference type supplied"));
+    }
+
+    private Validation<String, String> validateCcdCaseRef(Object caseRef) {
+        return Optional.of(caseRef)
+            .filter(ref -> ref instanceof String)
+            .map(ref -> ((String) ref).replaceAll("[^0-9]", ""))
+            .filter(ref -> !Strings.isNullOrEmpty(ref))
```

Comment:
I know it's not new code, but... is it me, or is this validation not only unclear but also wrong?
Perhaps worth adding a todo comment and/or jira ticket.